devtools/replay/RecordingState.jsm

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+ChromeUtils.defineModuleGetter(
+  this,
+  "Services",
+  "resource://gre/modules/Services.jsm"
+);
+
+const env = Cc["@mozilla.org/process/environment;1"].getService(
+  Ci.nsIEnvironment
+);
+
+const RECORDING_STATE = {
+  ERROR: 0,
+  READY: 1,
+  STARTING: 2,
+  RECORDING: 3,
+  STOPPING: 4,
+};
+
+class RecordingState {
+  constructor (start, stop) {
+    this.recordingMap = new WeakMap();
+    this.startCallback = start;
+    this.stopCallback = stop;
+  }
+
+  // Return whether all tabs are automatically being recorded.
+  isRecordingAllTabs() {
+    return env.get("RECORD_ALL_CONTENT")
+        || Services.prefs.getBoolPref("devtools.recordreplay.alwaysRecord");
+  }
+
+  toggle (browser, button) {
+    const obj = this.recordingMap.get(browser);
```

Comment:
Could you verify, if you drag a tab into a new window, does the `browser` element stay the same? I'm not certain that `browser` is a persistent key we can rely on in the map, but it might be, I'm just unsure.

Reply:
I'll check this out. I originally used a `Map` with the `pid` as the key but that didn't seem persistent either.

Reply:
The browser isn't a suitable key, unfortunately. I'll try `pid` again and see if I can sort out why that seems to work elsewhere but not for my use case.

Reply:
I don't think it is safe to rely on PID for this case because the process can crash and respawn with the same ID.

If we didn't have the `starting` state for now, the `stopping` state could probably be implemented with out it, I think?

Reply:
Can we hook window creation/destruction and generate our own IDs at attach them as an attribute? It seems like there's value in consistently identifying a tab but maybe I'm overthinking it.